src/main/java/com/accounts/transfer/solutions/one/TwoAccountsBlockingTransferExecutor.java

```diff
@@ -0,0 +1,49 @@
+package com.accounts.transfer.solutions.one;
+
+import com.accounts.model.Account;
+import com.accounts.storage.IncorrectRequstedAccountIdException;
+import com.accounts.transfer.AccountValidator;
+import com.accounts.transfer.TransferExecutor;
+import com.accounts.transfer.TransferFailureException;
+import com.google.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+
+import java.math.BigInteger;
+import java.util.TreeSet;
+
+@Slf4j
+public class TwoAccountsBlockingTransferExecutor implements TransferExecutor {
+
+    private AccountValidator validator;
+
+    @Inject
+    public TwoAccountsBlockingTransferExecutor(AccountValidator validator) {
+        this.validator = validator;
+    }
+
+    @Override
+    public void execute(Account giver, Account taker, BigInteger amount) throws TransferFailureException {
+        TreeSet<Account> accounts = sortAccountsInLockingOrder(giver, taker);
+        log.info("from: {} to: {} money to transfer: {}, transfer STARTED", giver, taker, amount);
+        synchronized (accounts.pollFirst()){
+            synchronized (accounts.pollFirst()){
```

Comment:
Too complex line.
Refactor it to several separate lines.

Reply:
which one?